sdk/src/main/java/io/dapr/client/DaprClientGrpc.java

```diff
@@ -182,6 +189,82 @@ public Mono<Void> publishEvent(PublishEventRequest request) {
     }
   }
 
+  /**
+   *
+   * {@inheritDoc}
+   */
+  public <T> Mono<BulkPublishResponse> publishEvents(BulkPublishRequest<T> request) {
+    try {
+      String pubsubName = request.getPubsubName();
+      String topic = request.getTopic();
+      DaprProtos.BulkPublishRequest.Builder envelopeBuilder = DaprProtos.BulkPublishRequest.newBuilder();
+      envelopeBuilder.setTopic(topic);
+      envelopeBuilder.setPubsubName(pubsubName);
+
+      for (BulkPublishRequestEntry<?> entry: request.getEntries()) {
+        Object event = entry.getEvent();
+        byte[] data;
+        String contentType = entry.getContentType();
+
+        // Serialize event into bytes
+        if (!Strings.isNullOrEmpty(contentType) && objectSerializer instanceof DefaultObjectSerializer) {
```

Comment:
This is a problem we need to fix in other places in the SDK. Using the default serializer should not trigger different behavior. It should be based on the content-type only.

Reply:
this is a change that needs to be done SDK wide and not only for one api ... That might cause serialized events from this API to become incompatible with the other normal publish API ...